src/Components/Web/src/Routing/NavLink.cs

```diff
@@ -44,6 +44,12 @@ public class NavLink : ComponentBase, IDisposable
     [Parameter]
     public RenderFragment? ChildContent { get; set; }
 
+    /// <summary>
+    /// Gets or sets a flag to indicate whether route matching should ignore the query string.
+    /// </summary>
+    [Parameter]
+    public bool IgnoreQueryString { get; set; }
```

Comment:
Hmm, I wonder if this behavior could potentially be treated as a subset of https://github.com/dotnet/aspnetcore/issues/18163?

Reply:
I don't think we need a new flag for this. This is just a bug/oversight in existing code

Reply:
> I don't think we need a new flag for this

Are you referring to the whole `IgnoreQueryString` flag, or to some other hypothetical new flag related to @TanayParikh's comment about the hash part of the URL?

If you mean `IgnoreQueryString` itself, I agree it would be better if we were ignoring the querystring by default. However that would be a breaking change we would need to weigh up carefully.

Reply:
I don't think matching on the query string would have offered a good behavior before. Query strings are not ordered, so the chances of matching on the query string would have been incredibly narrow.

I think we can just make the change and make an announcement about it if we deeply care about it.

Reply:
I agree about the default, however there are clear cases where someone might have wanted it to match on querystrings and might in fact be depending on that behavior now. For example, if you have:

```razor
@page "/products"

@code {
    [Parameter]
    [SupplyParameterFromQuery(Name = "category_id")]
    public int CategoryId { get; set; }
}
```

... then you might also have navigation links like:

```razor
<NavLink href="products?category_id=1">Dogs</NavLink>
<NavLink href="products?category_id=2">Cats</NavLink>
<NavLink href="products?category_id=3">Fish</NavLink>
```

... and expect those links to light up to show the current page.

It's not the most common pattern, but it's a valid and believable one so we should be prepared for making a clear announcement and treating it as a breaking change if we're going to change this.

Reply:
Yep, I'm fine if we do that.

The alternative is to properly support query string parameters, but that is much more work.

I want to avoid the situation where we have a behavior that "half" works and I think in this case it is simpler to not support it at all vs to add full support for it.